### PR TITLE
`std::dims`

### PR DESCRIPTION
**Pull request by fbusato**
Address https://github.com/NVIDIA/cccl/issues/2810

## Description

Add C++26 `std::dims` alias template backport to C++17 (aligned with `mdspan`)  https://en.cppreference.com/w/cpp/container/mdspan/extents

### CODE REVIEW

**Review by pciolkosz — APPROVED** [review]
I know this is a simple alias, but should there be at least one small test case that will use it for something trivial, just to make sure it's available, not under wrong ifdef etc?

**Comment by fbusato** [comment]
good point. I will do that

File: libcudacxx/test/libcudacxx/std/containers/views/mdspan/mdspan.extents.dims/compare.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
```
Comment (miscco) [inline libcudacxx/test/libcudacxx/std/containers/views/mdspan/mdspan.extents.dims/compare.pass.cpp:6]:
```suggestion
// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
```

File: libcudacxx/test/libcudacxx/std/containers/views/mdspan/mdspan.extents.dims/compare.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+int main(int, char**)
+{
+  {
+    using index_t = size_t;
+
+    cuda::std::dextents<index_t, 3> e0{1, 2, 3};
+    cuda::std::dims<3> e1{1, 2, 3};
+
+    assert(e0 == e1);
```
[the remark this answers is not in the document]
Comment (miscco) [reply libcudacxx/test/libcudacxx/std/containers/views/mdspan/mdspan.extents.dims/compare.pass.cpp:24]: Yeah we should do that
```suggestion
    cuda::std::dextents<index_t, 3> e0{1, 2, 3};
    cuda::std::dims<3> e1{1, 2, 3};

    static_assert(cuda::std::is_same<decltype(e0), decltype(e1)>::value, "");
    assert(e0 == e1);
```

**Review by miscco — CHANGES_REQUESTED** [review]

**Review by miscco — APPROVED** [review]

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**